x-pack/plugins/enterprise_search/server/lib/crawler/post_crawler_multiple_schedules.ts

```diff
@@ -0,0 +1,44 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import { IScopedClusterClient } from '@kbn/core/server';
+
+import { CONNECTORS_INDEX } from '../..';
+import { Connector } from '../../../common/types/connectors';
+
+export const postCrawlerCustomScheduling = async (
+  client: IScopedClusterClient,
+  indexName: string,
+  customSchedules: Object
+): Promise<Connector | undefined> => {
+  console.log('Inside postCrawlerCustomScheduling');
+  console.log(customSchedules);
+
+  const connectorId = await fetchCrawlerDocumentIdByIndexName(client, indexName);
+  const crawlerResult = await client.asCurrentUser.update<Connector>({
+    index: CONNECTORS_INDEX,
+    id: connectorId,
+    doc: {
+      custom_scheduling: Object.fromEntries(customSchedules),
+    },
+  });
+  console.log(crawlerResult);
+  return crawlerResult;
+};
+
+export const fetchCrawlerDocumentIdByIndexName = async (
```

Comment:
This should have its own file. Please refer to [Server Guidelines](https://github.com/elastic/kibana/blob/main/x-pack/plugins/enterprise_search/SERVER.md)